src/components/designSystem/ButtonLink.tsx

```diff
@@ -0,0 +1,127 @@
+import { forwardRef, ReactNode, MouseEvent } from 'react'
+import { Link } from 'react-router-dom'
+import styled, { css } from 'styled-components'
+
+import { theme } from '~/styles'
+
+import { Icon, IconName } from './Icon'
+import { Typography } from './Typography'
+
+export interface ButtonLinkProps {
+  to: string
+  title?: string | number
+  icon?: IconName
+  active?: boolean
+  disabled?: boolean
+  external?: boolean
+  onClick?: (e: MouseEvent<HTMLAnchorElement>) => void
+}
+
+interface ConditionalWrapperProps extends Pick<ButtonLinkProps, 'external' | 'to'> {
+  $active?: boolean
+  $disabled?: boolean
+  children: ReactNode
+  onMouseDown: (e: MouseEvent<HTMLAnchorElement>) => void
+}
+
+const ConditionalWrapper = forwardRef<HTMLAnchorElement, ConditionalWrapperProps>(
+  ({ external, to, children, ...props }: ConditionalWrapperProps) => {
+    return external ? (
+      <ExternalLink href={to} {...props}>
```

Comment:
For external link should we force by default rel="noopener noreferrer" ? 
And target="_blank" could be given during invocation in case we do not want to open in new tab in some cases WDYT ?